File: tax_identifier.py
#import entrez module
import sys
from Bio import Entrez

# get arguments
args = sys.argv[1:]

# check input arguments
if len(args) < 1:
	print "Expect at least one tax id as an input argument"
	sys.exit()

# set variables
#taxids = [515482, 515474]
taxids = args

# set email
Entrez.email = "gawbul@gmail.com"

# traverse ids
for taxid in taxids:
	handle = Entrez.efetch(db="taxonomy", id=taxid, mode="text", rettype="xml")
	records = Entrez.read(handle)
	for taxon in records:
		taxid = taxon["TaxId"]
		name = taxon["ScientificName"]
		tids = []
		for t in taxon["LineageEx"]:
			tids.insert(0, t["TaxId"])
		tids.insert(0, taxid)
		print "%s\t|\t%s\t|\t%s" % (taxid, name, " ".join(tids))

File: tax_identifier.pl
use strict;
use warnings;
use Bio::DB::Taxonomy;
use Bio::Tree::Tree;

my @args = @ARGV;

if (scalar(@args) < 1) {
	die("Expect at least one input argument\n");
}

#my @taxonids = (515482, 515474);
my @taxonids = @args;
my @lineages = ();

my $db = Bio::DB::Taxonomy->new(-source => 'entrez');

foreach my $taxonid (@taxonids) {
	my $taxon = $db->get_taxon(-taxonid => $taxonid);
	my $tree = Bio::Tree::Tree->new(-node => $taxon);
	my @taxa = $tree->get_nodes;
	my @tids = ();
	foreach my $t (@taxa) {
		unshift(@tids, $t->id());
	}
	push(@lineages, $taxonid . "\t|\t" . $taxon->scientific_name() . "\t|\t" . "@tids");
}

foreach my $lineage (@lineages) {
	print "$lineage\n";
}
